cpu/esp32/periph/uart.c

```diff
@@ -187,40 +180,17 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
 
 void uart_poweron (uart_t uart)
 {
-    switch (uart) {
-        #if UART_NUMOF
-        case 0:  periph_module_enable(PERIPH_UART0_MODULE);
-                 _uart_config(uart);
-                 break;
-        #endif
-        #if UART_NUMOF > 1
-        case 1:  periph_module_enable(PERIPH_UART1_MODULE);
-                 _uart_config(uart);
-                 break;
-        #endif
-        #if UART_NUMOF > 2
-        case 2:  periph_module_enable(PERIPH_UART2_MODULE);
-                 _uart_config(uart);
-                 break;
-        #endif
-        default: break;
-    }
+    CHECK_PARAM (uart < UART_NUMOF);
```

Comment:
Maybe just replace with an assert, this goes for all `CHECK_PARAM`

Reply:
@MrKevinWeiss This is not only the case for `periph/uart` but for other ESP32/ESP8266 peripherals and other modules. I have it on my ToDo list to change it. Maybe we could leave it for the moment to keep consistency with other modules. But, if you think it would be reasonable to change it with this PR, it would not be a big thing to change it. As you want.

Reply:
If there is a plan to do an update later that is fine then.